docs/developer/quickstart.md

```diff
@@ -10,6 +10,17 @@ Follow these steps to set up your development environment.
 
 1. Install:
     - [Go](https://golang.org/doc/install)
+
+      ```shell
+      # `go.mod` requires Go 1.21.3.
```

Comment:
I am unsure whether we should provide detailed instructions like these, or simply state that NGF requires Go 1.21.3. I use the same commands in the open-source project that I maintain.

Reply:
The latest version is 1.21.4 which is not compatible with the current `go.mod`, as seen on https://go.dev/dl/. 

1. Will it be helpful to add a sentence advising contributors to refer to `go.mod` as the source of truth instead of saying "the latest version of Go"?
2. Should this PR include the installation instructions or do you prefer to not include them?

Thanks!

Reply:
As long as the Go version you use to build is equal to or greater than what's in go.mod, you should be fine. You just can't be using a version of Go that's less than what's in go.mod.

This is why I say use "latest", because it'll always be equal to or greater than what's in go.mod.